src/contrib/cluster/Akka.Cluster.Sharding/DDataShardCoordinator.cs

```diff
@@ -405,7 +408,7 @@ bool Receive(object message)
                         return true;
 
                     case RememberEntitiesCoordinatorStore.UpdateDone m:
-                        if (!shardId.Contains(m.ShardId))
+                        if (shardId != null && !shardId.Equals(m.ShardId))
```

Comment:
These checks are the key bug fixes - enabling nullability made them easy to spot.